app/main.py

```diff
@@ -2,5 +2,21 @@
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    storage = {}
+
+    def wrapper(*args) -> Callable:
+        try:
+            key = tuple(args)
+        except TypeError as e:
+            raise ValueError(f"All arguments must be hashable: {e}")
```

Comment:
The current implementation catches a `TypeError` if any argument is not hashable and raises a `ValueError`. While this handles the error, it would be better to ensure that only immutable types are used as keys. Consider converting mutable arguments to immutable ones, such as using a tuple of tuples for lists or dictionaries, to avoid this issue.